src/org/omegat/gui/glossary/GlossarySearcher.java

```diff
@@ -131,9 +131,19 @@ private boolean isTokenMatch(Token[] fullTextTokens, String fullText, String ter
                 Preferences.GLOSSARY_NOT_EXACT_MATCH_DEFAULT);
         List<Token[]> foundTokens = DefaultTokenizer.searchAll(fullTextTokens, glosTokens, notExact);
         foundTokens.removeIf(toks -> !keepMatch(toks, fullText, term));
+        foundTokens.removeIf(toks -> !rawMatch(toks, fullText, term));
         return foundTokens;
     }
 
+    private static boolean rawMatch(Token[] tokens, String srcTxt, String term) {
+        for (Token token : tokens) {
+            if (token.getTextFromString(srcTxt).equals(term.substring(token.getLength()))) {
```

Comment:
Do I understand this right?

- `token.getTextFromString(srcTxt)` is the text that was matched in the source text
- `term.substring(token.getLength())` is a prefix of the original term

You are rejecting any matched term where none of the tokens is a prefix of the term.

This makes some severe assumptions about how stemming/lemmatizing works. I don't think the token is guaranteed to be a prefix, or even necessarily present in the original source string.

Reply:
> This makes some severe assumptions about how stemming/lemmatizing works. I don't think the token is guaranteed to be a prefix, or even necessarily present in the original source string.

I'm also afraid this but I don't know well about stemming/lemmatizing behavior.
Actually I do only on CJK  in my first trial implementation.
IMHO, the assumption is OK on CJK languages.

Reply:
>  IMHO, the assumption is OK on CJK languages.

I think either we need some sort of positive confirmation that it would work with all languages, or to limit this change to only CJK.

Reply:
Because the issue, hash collision,  is reported on Japanese, it would be better to limit the change only to CJK.

Reply:
I only just realized that our `isCJK` method is very, very stupid. I'm now worried that this will apply to many more cases than it should have.